# Creating Groups with rules

In [1]:
import pandas as pd

In [2]:
user_features= pd.read_csv('user_features_data/user_features_cleaned.csv')

groups I want to create:
- high spenders
- maybe discount hunters? could be too small
- no bookers (gets also rid of all the NaNs in the data)
- seniors married/single?
- young & single?
- people without return flight?

In [3]:
#calculate total average revenue -> similar to average per day total?

In [4]:
user_features.describe().T

,count,mean,std,min,25%,50%,75%,max
user_id,5782.0,547670.236077,64035.394540,94883.000000,519413.750000,542279.500000,576215.500000,844489.00000
Age,5782.0,43.887928,12.037172,19.000000,37.000000,44.000000,51.000000,91.00000
days_in,5782.0,1270.738499,34.194997,1154.000000,1255.000000,1272.000000,1283.000000,1619.00000
trip_session_ratio,5274.0,2.808584,1.613452,1.000000,1.750000,2.333333,3.458333,10.00000
no_cancellations,5782.0,0.106192,0.318055,0.000000,0.000000,0.000000,0.000000,2.00000
ave_page_clicks,5782.0,17.646602,8.612828,4.125000,12.555556,15.625000,19.750000,109.12500
ave_discount,5782.0,0.072654,0.078744,0.000000,0.000000,0.050000,0.125000,0.50000
nights_median,5150.0,4.126214,2.734033,1.000000,2.500000,3.500000,5.000000,30.00000
hotel_avg_rev,5150.0,947.664624,952.234480,38.000000,426.162500,692.000000,1106.937500,14057.00000
hotels_no_bookings,5180.0,2.694402,1.340564,1.000000,2.000000,3.000000,4.000000,8.00000


## High spenders
no. 1 priority -> more money to make

In [ ]:
group_high_spenders =
rest_users =

## no bookers
chose this for nr two priority, so that I don't have any nan values in the other groups for later analysis
Also, they booked nothing -> more work to get them to book than the others.

In [5]:
user_features[["flight_no_bookings", "hotels_no_bookings"]].isna().value_counts()

flight_no_bookings  hotels_no_bookings
False               False                 4797
True                True                   508
                    False                  383
False               True                    94
Name: count, dtype: int64

- 4797 booked both - hotel & flight
- 508 booked nothing
- 383 booked only hotel
- 94 booked only flights

In [10]:
group_no_bookers = rest_users[rest_users["flight_no_bookings"].isna() & rest_users["hotels_no_bookings"].isna()].copy()

In [11]:
rest_users = user_features[~user_features.index.isin(group_no_bookers.index)]

In [12]:
group_only_hotel_bookers = rest_users[rest_users["flight_no_bookings"].isna()].copy()
rest_users = rest_users[~rest_users.index.isin(group_only_hotel_bookers.index)]

In [13]:
group_only_flight_bookers = rest_users[rest_users["hotels_no_bookings"].isna()].copy()
rest_users = rest_users[~rest_users.index.isin(group_only_flight_bookers.index)]